oximeter/db/schema/replicated/5/up.sql

```diff
@@ -0,0 +1,279 @@
+ALTER TABLE oximeter.measurements_histogrami8_local ON CLUSTER oximeter_cluster ADD COLUMN min Int8;
```

Comment:
I forgot that we made all scalar measurement columns nullable in version 4, to support missing samples. We did _not_ do that for the histogram types, because arrays cannot be nullable in ClickHouse, e.g., you can't have a type `Nullable(Array(Int8))`. Since we know that all `Histogram`s are non-empty by construction (they have at least one bin), then we use an empty array to represent those missing samples.

So, what do we do here? For the scalar columns, `max`, `sum_of_samples` and `squared_mean`, I think we want them to be `Nullable(T)`. For their defaults, I think it unfortunately depends on whether the sample is missing or not (whether the bins / counts arrays are empty). I _think_ you can write that like this:

```
DEFAULT if(empty(bins), NULL, 0)
```

I would do the same thing for the array columns, but using `if(empty(bins), [], [0, 0, 0, 0, 0])` instead.

There is actually one other option, which I think might be better. We could just assert that it's fraught to set a default on these columns, even in the interest of preserving the existing data. Instead, one could drop those tables and recreate them with the desired structure. I just double-checked, and the only existing histogram timeseries are the Nexus request latencies, which are not consumed anywhere at all. You could have one SQL file which drops each table and one which recreates it, this time _without_ defaults. (Those are only there to support the existing data, which we're dropping.)

Reply:
As discussed, will do a drop/create series for these.